shell/common/rasterizer.cc

```diff
@@ -164,6 +169,13 @@ void Rasterizer::DoDraw(std::unique_ptr<flutter::LayerTree> layer_tree) {
         ScreenshotLastLayerTree(ScreenshotType::SkiaPicture, false);
     persistent_cache->DumpSkp(*screenshot.data);
   }
+
+  // TODO(liyuqian): in Fuchsia, the rasterization doesn't finish when
+  // Rasterizer::DoDraw finishes. Future work is needed to adapt the timestamp
+  // for Fuchsia.
```

Comment:
What extra work after |Rasterizer::DoDraw| are you referring to specifically?  Are you talking about the GPU work (as opposed to the CPU "GPU thread" work in Flutter)?  What does the future work you mention here look like?  In some Fuchsia traces I had lying around, the GPU thread's work (covered by the "GPURasterizer::Draw") does nearly end with |Rasterizer::DoDraw| (there's just a couple of microseconds after it).  If I'm understanding this correctly, I'd be happy to clarify that this is just Flutter GPU thread work here, and have the burden of measuring GPU work be on Flutter runner.

Reply:
You're right that `Rasterizer::DoDraw` and `Rasterizer::Draw` (which adds `GPURasterizer::Draw` tracing) are approximately equal. The extra work I referred was `SceneUpdateContext::ExecutePaintTasks` in [scene_update_context.cc](https://github.com/flutter/engine/blob/master/flow/scene_update_context.cc). I've added this info to the code comment.